api/allIncidents/adminIncidentsRouter.js

```diff
@@ -0,0 +1,156 @@
+const express = require('express');
+const router = express.Router();
+const Incidents = require('./incidentsModel');
+const {
+  validateAndSanitizeIncidentQueries,
+  authRequired,
+  checkIncidentExists,
+  validateIncident,
+} = require('../middleware');
+
+// TODO document shape of objects coming and going
+
+router.use(authRequired);
+
+/**
+ * @swagger
+ * /:
+ *  GET:
+ *    Summary: Path returning all incidents in reverse chronological order and filtered according to queries  *      {
+ *        state as string,
+ *        startDate as integer,
+ *        endDate as integer,
+ *        limit as integer
+ *      }
+ *
+ *    tags:
+ *      - incidents
+ *    produces:
+ *      - application/json
+ *    responses:
+ *      200:
+ *        description: Success ... returns an array of incident objects
+ *      500:
+ *        description: Server response error
+ */
+
+router.get('/', validateAndSanitizeIncidentQueries, async (req, res, next) => {
+  const sanitizedQueries = req.sanitizedQueries;
+
+  Incidents.getIncidents(sanitizedQueries)
+    .then((incidents) => {
+      res.status(200).json(incidents);
+    })
+    .catch(next);
+});
+
+/**
+ * @swagger
+ * /{incident_id}:
+ *  GET:
+ *    Summary: Path returning single incident by incident_id
+ *    tags:
+ *      - incidents
+ *    produces:
+ *      - application/json
+ *    responses:
+ *      200:
+ *        description: Success ... returns incident object
+ *      500:
+ *        description: Server response error
+ */
+
+router.get('/:incident_id', checkIncidentExists, async (req, res) => {
+  res.status(200).json(req.incident);
+});
+
+/**
+ * @swagger
+ * /{incident_id}:
+ *  PUT:
+ *    Summary: Path updating single incident by incident_id
+ *    tags:
+ *      - incidents
+ *    produces:
+ *      - application/json
+ *    responses:
+ *      201:
+ *        description: Success ... returns updated incident object
+ *      500:
+ *        description: Server response error
+ */
+
+router.put(
```

Comment:
is there a way to get this formatted like the other routes? just nitpicking

Reply:
Formatted in what sense? Using Try Catch?

Reply:
just have them inline instead of cascading, but I think that has to do with the formatter

Reply:
You mean the then/catch?